aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -748,10 +748,13 @@ Tensor frobenius_norm(const Tensor& self, IntArrayRef dim, bool keepdim) {
   if (dim.size() == 1 || dim.size() == 0) {
     return at::norm(self, 2, dim, keepdim);
   }
+  auto dim_ = dim.vec();
```

Comment:
Most functions with variants just construct an out tensor placehold and then call the out variant. Why doesn't frobenius norm follow this pattern? That is, why doesn't frobenious norm just do:

```
Tensor result;
return frobenius_norm_out(result, ...)
```

?

Reply:
Not sure why frobenius and nuclear were implemented like this. I can refactor them. It would be better to do that in a followup PR, right?

Reply:
Not refactoring them in this PR sounds good.